src/doc/book/ffi.md

```diff
@@ -526,10 +526,10 @@ However, the language provides a workaround.
 
 As a special case, an `enum` is eligible for the "nullable pointer optimization" if it
 contains exactly two variants, one of which contains no data and the other contains
-a single field of one of the non-nullable types listed above. This means it is represented
-as a single pointer, and the non-data variant is represented as the null pointer. This is
-called an "optimization", but unlike other optimizations it is guaranteed to apply to
-eligible types.
+a single field of one of the non-nullable types listed above (or a struct containing such a type).
```

Comment:
Don't think it's limited to a single field either. Just there's a non-nullable field somewhere in there, even nested.